### PR TITLE
MM-69074 Added nil guard on webhook handler (#501)

### PR DESCRIPTION
**Pull request by avasconcelos114**
#### Summary

Cherry-pick PR of #501 

## Change Impact: 🟢 Low

**Reasoning:** This is a defensive nil guard addition to the webhook handler that prevents potential nil pointer dereferences when Microsoft Graph sends null entries in webhook payloads. The change is isolated to a single function and does not modify behavior for valid inputs. Comprehensive test coverage covering null entries, mixed payloads, and validation flows significantly reduces regression risk.

**Regression Risk:** Very low. The nil guard only adds defensive logic to skip malformed entries and log them—behavior that previously would have caused a panic. Valid webhook entries are processed identically to before. The change touches no shared dependencies, public APIs, or critical business logic flows. It's a cherry-pick of an already-reviewed PR `#501`.

**QA Recommendation:** Minimal manual QA required. The new test suite comprehensively covers the added nil-guard behavior, validation token handling, and edge cases (empty arrays, mixed null/valid entries, invalid JSON). Focus QA on integration testing the webhook endpoint with malformed payloads from Microsoft Graph to ensure graceful handling in production and that null entries don't block processing of subsequent valid entries.

### CODE REVIEW

File: msgraph/handle_webhook_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+			req := httptest.NewRequest(http.MethodPost, "/notification/v1/event", strings.NewReader(tc.body))
+			rec := httptest.NewRecorder()
+
+			require.NotPanics(t, func() {
+				notifications := remote.HandleWebhook(rec, req)
+				require.Len(t, notifications, tc.wantNotificationLen)
+				if tc.wantSubscriptionID != "" {
+					require.Equal(t, tc.wantSubscriptionID, notifications[0].SubscriptionID)
+				}
+			})
```
[the remark this answers is not in the document]
Comment (avasconcelos114) [reply msgraph/handle_webhook_test.go:84]: Considering this is a cherry-pick of a PR already merged to master, I would refrain from making changes to test suites here

**Comment by avasconcelos114** [comment]
@coderabbitai approve

**Review by nang2049 — APPROVED** [review]
LGTM!
